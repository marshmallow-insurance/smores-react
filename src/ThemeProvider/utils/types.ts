import * as theme from '@mrshmllw/smores-foundations/build/web/variables.json'
import { Color } from 'theme'

type Flatten<T, Prefix extends string = ''> = {
  [K in keyof T & string]: T[K] extends Record<string, any>
    ? Flatten<T[K], `${Prefix}${K}.`>
    : `${Prefix}${K}`
}[keyof T & string]

export type NewColour = Flatten<(typeof theme)['color']>

const legacyColorMap = {
  marshmallowPink: 'interactive.primary.base', // old â†’ new
}

export type LegacyColorMap = keyof typeof legacyColorMap
