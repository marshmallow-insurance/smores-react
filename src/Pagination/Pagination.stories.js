import React from 'react'
import { Pagination } from './Pagination'

export default {
  title: 'Pagination',
  component: Pagination,
  argTypes: { handlePageChange: { action: 'pageChanged' } },
}

const Template = (args) => <Pagination {...args} />

export const Default = Template.bind({})

Default.args = {
  total: 7,
  partition: 1,
}

export const LongerPagination = Template.bind({})

LongerPagination.args = {
  total: 11,
  partition: 1,
}
