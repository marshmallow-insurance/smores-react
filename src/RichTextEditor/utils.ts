import { $createParagraphNode, LexicalNode, RootNode } from 'lexical'

export const appendNodes = (root: RootNode, nodes: LexicalNode[]) => {
  nodes
    .filter((node) => node.__type !== 'linebreak')
    .map((node) => {
      if (node.__type === 'text') {
        const paragraphNode = $createParagraphNode()
        paragraphNode.append(node)
        return paragraphNode
      }
      return node
    })
    .forEach((node) => root.append(node))
}
