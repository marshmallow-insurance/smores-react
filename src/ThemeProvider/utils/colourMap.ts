import * as designTokens from '@mrshmllw/smores-foundations/build/web/variables.json'
import { Color } from 'theme'

type Flatten<T, Prefix extends string = ''> = {
  [K in keyof T & string]: T[K] extends Record<string, any>
    ? Flatten<T[K], `${Prefix}${K}.`>
    : `${Prefix}${K}`
}[keyof T & string]

export type NewColor = Flatten<
  Pick<typeof designTokens, 'color' | 'extended1' | 'thirdParty'>
>

export type ColorTypes = Color | NewColor

// TODO: Test how well this works
// we currently access numerical colours like `theme.color.neutral[100]` or in the case of 000 colours, `theme.color.neutral['000']`
export const getThemeColor = (path: NewColor): string => {
  return path
    .split('.')
    .reduce((acc, key) => acc?.[key], designTokens as any) as string
}

// a function that returns a flattened dot notation string path to access the color value
export const formatDesignTokenColor = (): string => {
  const { color, extended1, thirdParty } = designTokens
  const colors = { color, extended1, thirdParty }

  if (colors === null) {
    return ''
  }

  const result: string[] = []
  const destructureNestedObject = (obj: any, currentPath: string) => {
    for (const key in obj) {
      if (Object.prototype.hasOwnProperty.call(obj, key)) {
        const value = obj[key]
        const newPath = currentPath ? `${currentPath}.${key}` : key
        if (typeof value === 'object' && value !== null) {
          destructureNestedObject(value, newPath)
        } else {
          result.push(newPath)
        }
      }
    }
  }
  destructureNestedObject(colors, '')
  return result.join(', ')
}

// old colour name â†’ new colour path // base token value
export const legacyColorMap: Record<Color, NewColor> = {
  fairyFloss: 'color.surface.brand.100', // palette.primary.040
  bubblegum: 'color.surface.brand.200', // palette.primary.060
  marshmallowPink: 'color.surface.brand.300', // palette.primary.100
  lollipop: 'color.surface.brand.400', // palatte.primary.120

  // Core Secondary
  chia: 'color.feedback.inactive.100', // palette.secondary.040
  sesame: 'color.text.subtle', // palette.secondary.060
  liquorice: 'color.text.base', // palette.secondary.100
  boba: 'color.text.contrast', // palette.secondary.120

  // Core Tertiary
  cream: `color.surface.base.000`, // palette.neutral.000
  coconut: 'color.surface.base.100', // palette.neutral.010
  mascarpone: 'color.surface.base.200', // palette.neutral.020
  custard: 'color.surface.base.300', // palette.neutral.040

  // Brand Secondary
  feijoa: 'color.illustration.accent1.100', // palette.brand1.060
  spearmint: 'color.illustration.accent1.200', // palette.brand1.100
  macaroon: 'color.illustration.accent2.100', // palette.brand2.060
  blueberry: 'color.illustration.accent2.200', // palette.brand2.100
  pistachio: 'color.illustration.accent3.200', // palette.brand3.100
  matcha: 'color.illustration.accent3.100', // palette.brand3.060
  caramel: 'color.illustration.accent4.200', // palette.brand4.100
  peanut: 'color.illustration.accent4.100', // palette.brand4.060
  marzipan: 'color.illustration.neutral.400', // palette.neutral.100
  oatmeal: 'color.illustration.neutral.300', // palette.neutral.060
  satsuma: 'extended1.20', // extended1.020

  // Traffic light
  watermelon: 'color.feedback.negative.100', // palette.negative.020
  strawberry: 'color.feedback.negative.200', // palette.negative.100
  apple: 'color.feedback.positive.200', // palette.positive.100
  mint: 'color.feedback.positive.100', // palette.positive.020
  lemon: 'color.feedback.notice.200', // palette.notice.100
  sherbert: 'color.feedback.notice.100', // palette.notice.020
  tangerine: 'extended1.100', // extended1.100

  // Third-party brand colours
  compareTheMarket: 'thirdParty.compareTheMarket',
  confused: 'thirdParty.confusedCom',
  onfido: 'thirdParty.onfido',
  x: 'thirdParty.twitter', // x rebrand not yet reflected in foundations
  premfina: 'thirdParty.premfina',
  checkout: 'thirdParty.checkout',
  meta: 'thirdParty.facebook', // Meta rebrand not yet reflected in foundations
  stripe: 'thirdParty.stripe',
  intercom: 'thirdParty.intercom',
  ravelin: 'thirdParty.ravelin',
  rac: 'thirdParty.rac',
  hometree: 'thirdParty.hometree',
} as const

/* usage example */
// prop change:
// change color prop type to be `color?: Color | NewColor`
// component usage:
// <Component color={'marshmallowPink'}/>
// OR
// <Component color={'interactive.primary.base'}/>
// inside component:
// const resolvedColor = color in legacyColorMap ? getThemeColor(legacyColorMap[color]) : getThemeColor(color)

// the above should resolve to #FF88C8
