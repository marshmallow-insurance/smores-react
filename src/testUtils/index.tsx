import React, { ReactElement } from 'react'
import { render, RenderOptions } from '@testing-library/react'
import { SmoresThemeProvider } from '../ThemeProvider/ThemeProvider'

const Providers = ({ children }: { children: React.ReactNode }) => {
  return <SmoresThemeProvider>{children}</SmoresThemeProvider>
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>,
) => render(ui, { wrapper: Providers, ...options })

export * from '@testing-library/react'
export * from '@testing-library/jest-dom/vitest'
export * from 'jest-styled-components'

export { customRender as render }
