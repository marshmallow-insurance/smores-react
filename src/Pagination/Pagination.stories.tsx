import React from 'react'
import { Pagination, PaginationProps } from './Pagination'

export default {
  title: 'Pagination',
  component: Pagination,
  argTypes: { handlePageChange: { action: 'pageChanged' } },
}

const Template = (props: PaginationProps) => <Pagination {...props} />

export const Default = Template.bind({})

Default.args = {
  total: 7,
  partition: 1,
}

export const LongerPagination = Template.bind({})

LongerPagination.args = {
  total: 11,
  partition: 1,
}
