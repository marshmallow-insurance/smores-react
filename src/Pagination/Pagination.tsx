import React, { FC, useEffect, useState } from 'react'
import styled from 'styled-components'

import { Box } from '../Box'
import { Text } from '../Text'
import { Icon } from '../Icon'

import { theme } from '../theme'
import { MarginProps } from '../utils/space'

const MAX_PAGES = 7
const MAX_ADDITIONAL_PAGES = Math.ceil(MAX_PAGES / 2)

export type PaginationProps = {
  /** Total number of records  */
  total: number
  /** Number of records per page  */
  partition: number
  /** Handle page change */
  handlePageChange: (page: number) => void
  /** Sets current page number */
  currentPage?: number
} & MarginProps

export const Pagination: FC<PaginationProps> = ({
  total,
  partition,
  handlePageChange,
  currentPage,
  ...marginProps
}) => {
  const [lastPage, setLastPage] = useState(0)
  const [activePage, setActivePage] = useState(1)
  const [pages, setPages] = useState<number[]>([])
  const [shownPages, setShownPages] = useState<number[]>([])
  const [showFirstDots, setShowFirstDots] = useState(false)
  const [showLastDots, setShowLastDots] = useState(false)

  useEffect(() => {
    // This is a rather hacky fix, in theory it should be listening to total, but because the total comes from the same endpoint as the data, it forces a reset to 1 every time
    // Using lastPage seems to bypass that
    if (currentPage) {
      setActivePage(currentPage)
    } else {
      setActivePage(1)
    }
  }, [lastPage, currentPage])

  useEffect(() => {
    const numberOfPages = Math.ceil(total / partition)
    if (numberOfPages > 1) {
      setLastPage(numberOfPages)
    } else if (numberOfPages === 1) {
      // Sets last page to be null to avoid the last page being kept from previous state.
      setLastPage(0)
    }
    if (numberOfPages > 2) {
      const allPages = Array.from({ length: numberOfPages }, (_, i) => i + 1)
      setPages(allPages.slice(1, numberOfPages - 1))
    } else {
      // Sets pages to empty to avoid using the previous state if pages set before.
      setPages([])
    }
  }, [total, partition])

  useEffect(() => {
    if (lastPage <= MAX_PAGES) {
      setShownPages(pages)
    } else if (activePage <= MAX_ADDITIONAL_PAGES) {
      setShownPages(pages.slice(0, MAX_ADDITIONAL_PAGES))
    } else if (activePage > lastPage - MAX_ADDITIONAL_PAGES) {
      setShownPages(pages.slice(pages.length - MAX_ADDITIONAL_PAGES))
    } else {
      const activePageIndex = pages.indexOf(activePage)
      setShownPages(pages.slice(activePageIndex - 1, activePageIndex + 2))
    }
  }, [lastPage, activePage, pages])

  useEffect(() => {
    setShowFirstDots(shownPages[0] !== pages[0])
    setShowLastDots(
      shownPages[shownPages.length - 1] !== pages[pages.length - 1],
    )
  }, [shownPages, pages])

  const movePage = (page: number) => {
    if (page > 0 && page <= lastPage) {
      setActivePage(page)
      handlePageChange(page)
    }
  }

  return (
    <Container flex direction="row" {...marginProps}>
      {activePage - 1 > 0 && (
        <PageBox onClick={() => movePage(activePage - 1)}>
          <Icon size={24} render="arrow" />
        </PageBox>
      )}
      <PageBox active={activePage === 1} onClick={() => movePage(1)}>
        <Text typo={activePage === 1 ? 'desc-medium' : 'desc-light'} tag="p">
          1
        </Text>
      </PageBox>
      {showFirstDots && (
        <Box flex justifyContent="center">
          <Text tag="p">...</Text>
        </Box>
      )}
      {shownPages.map((i) => {
        return (
          <PageBox
            active={activePage === i}
            key={i}
            onClick={() => movePage(i)}
          >
            <Text
              typo={activePage === i ? 'desc-medium' : 'desc-light'}
              tag="p"
            >
              {i}
            </Text>
          </PageBox>
        )
      })}
      {showLastDots && (
        <Box flex justifyContent="center">
          <Text tag="p">...</Text>
        </Box>
      )}
      {Boolean(lastPage) && (
        <PageBox
          active={activePage === lastPage}
          onClick={() => movePage(lastPage)}
        >
          <Text
            typo={activePage === lastPage ? 'desc-medium' : 'desc-light'}
            tag="p"
          >
            {lastPage}
          </Text>
        </PageBox>
      )}
      {activePage + 1 <= lastPage && (
        <PageBox onClick={() => movePage(activePage + 1)}>
          <Icon size={18} render="arrow" rotate={180} />
        </PageBox>
      )}
    </Container>
  )
}

interface IPageBox {
  active?: boolean
}

const Container = styled(Box)`
  align-items: center;
  height: 24px;
  > div {
    width: 24px;
    margin-right: 4px;

    &:last-child {
      margin-right: 0;
    }
  }
`

const PageBox = styled(Box)<IPageBox>`
  background: ${({ active }) => (active ? theme.colors.coconut : 'none')};
  cursor: pointer;
  display: flex;
  justify-content: center;
  border-radius: 50%;
`
