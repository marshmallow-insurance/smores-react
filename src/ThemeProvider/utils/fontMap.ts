import * as designTokens from '@mrshmllw/smores-foundations/build/web/variables.json'

import { Typo } from '../../Text'
import { Theme } from '../../ThemeProvider/ThemeProvider'
import { getFromObject } from '../../utils/getFromObject'

type Font = typeof designTokens.font
type FontKey = keyof Font
export type FontValueObject = Font[keyof Font][keyof Font[keyof Font]]

const fontValueKeys = [
  'fontFamily',
  'fontSize',
  'fontWeight',
  'lineHeight',
  'letterSpacing',
  'textCase',
]

type captionTypeKeys = [keyof Font['caption']]
type bodyTypeKeys = [keyof Font['body']]
type headingTypeKeys = [keyof Font['heading']]
type heroTypeKeys = [keyof Font['hero']]
type labelTypeKeys = [keyof Font['label']]
type linkTypeKeys = [keyof Font['link']]

export type FontTypeMap = {
  caption: captionTypeKeys
  label: labelTypeKeys
  body: bodyTypeKeys
  heading: headingTypeKeys
  hero: heroTypeKeys
  link: linkTypeKeys
}[keyof Font]

// type for the path to each font type in design tokens, e.g. 'font.hero.200'
export type FontPathMap = {
  [K in FontKey]: K extends 'caption'
    ? `font.${K}.${captionTypeKeys[number]}`
    : K extends 'body'
      ? `font.${K}.${bodyTypeKeys[number]}`
      : K extends 'heading'
        ? `font.${K}.${headingTypeKeys[number]}`
        : K extends 'hero'
          ? `font.${K}.${heroTypeKeys[number]}`
          : K extends 'label'
            ? `font.${K}.${labelTypeKeys[number]}`
            : never
}[FontKey]

export type TypoTypes = FontPathMap | Typo

// a function that returns a font value object based on either the new design token path or the legacy typo mapping
export const resolveToThemeFont = (
  typo: TypoTypes,
  theme: Theme,
): FontValueObject => {
  const fontPath = getTypoPath(typo) // always returns new design token path
  return getFromObject({
    obj: theme,
    path: fontPath,
  })
}

export const getTypoPath = (typo: TypoTypes) => {
  return typo in legacyFontStyleMapping
    ? legacyFontStyleMapping[typo as keyof typeof legacyFontStyleMapping]
    : typo
}

export const formatDesignTokenFont = (): string => {
  const { font } = designTokens
  if (font === null) {
    return ''
  }

  const result: string[] = []
  const destructureNestedObject = (obj: any, currentPath: string) => {
    for (const key in obj) {
      if (Object.prototype.hasOwnProperty.call(obj, key)) {
        const value = obj[key]

        const newPath = currentPath ? `${currentPath}.${key}` : key

        // TODO: instead of trimming, we should not add paths that end with font value keys
        const trimmedPath = fontValueKeys.some((key) =>
          newPath.endsWith(`.${key}`),
        )
          ? newPath.slice(0, newPath.lastIndexOf('.'))
          : newPath

        if (typeof value === 'object' && value !== null) {
          destructureNestedObject(value, trimmedPath)
        } else {
          result.push(trimmedPath)
        }
      }
    }
  }
  destructureNestedObject(font, 'font')
  return result.join(', ')
}

export const newFontStyles: Font = designTokens.font

export const legacyFontStyleMapping: Record<Typo, FontPathMap> = {
  'hero-alternate': 'font.hero.300',
  hero: 'font.hero.200',
  'heading-alternate': 'font.hero.100',
  'heading-large': 'font.heading.500',
  'heading-medium': 'font.heading.300',
  'heading-small': 'font.heading.200',
  'headline-regular': 'font.heading.100',
  'headline-small': 'font.heading.100', // query this
  'body-standfirst': 'font.body.300',
  'body-regular': 'font.body.200',
  'body-small': 'font.body.100',
  caption: 'font.caption.100',
  label: 'font.label.100',
}
