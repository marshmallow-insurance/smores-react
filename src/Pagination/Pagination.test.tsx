import React from 'react'
import { expect, it } from 'vitest'
import { render } from '../testUtils'
import { Pagination } from './Pagination'
import { noop } from '../utils/noop'

describe('Pagination', () => {
  it('renders correctly with default props', () => {
    const { baseElement } = render(
      <Pagination total={12} partition={1} handlePageChange={noop} />,
    )

    expect(baseElement).toMatchSnapshot()
  })

  it('renders correctly with current page prop', () => {
    const { baseElement } = render(
      <Pagination
        total={12}
        partition={1}
        handlePageChange={noop}
        currentPage={4}
      />,
    )

    expect(baseElement).toMatchSnapshot()
  })
})
