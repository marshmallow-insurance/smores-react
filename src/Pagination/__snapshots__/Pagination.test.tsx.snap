// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Pagination > renders correctly with current page prop 1`] = `
.c0 {
  display: flex;
}

.c3 {
  display: flex;
  align-items: center;
  justify-content: center;
  flex-shrink: 0;
  width: 18px;
  height: 18px;
  transform: rotate(0deg);
}

.c3 svg {
  color: #292924!important;
}

.c8 {
  display: flex;
  align-items: center;
  justify-content: center;
  flex-shrink: 0;
  width: 18px;
  height: 18px;
  transform: rotate(180deg);
}

.c8 svg {
  color: #292924!important;
}

.c7 {
  font-size: 14px;
  font-weight: 500;
  line-height: 20px;
  text-align: left;
  color: #292924;
}

.c5 {
  font-size: 16px;
  font-weight: 400;
  line-height: 20px;
  text-align: left;
  color: #292924;
}

.c1 {
  align-items: center;
}

.c1 >button {
  margin-right: 4px;
}

.c1 >button:last-child {
  margin-right: 0;
}

.c6 {
  background: #f1e9dc;
  cursor: pointer;
  display: flex;
  justify-content: center;
  border-radius: 24px;
  width: fit-content;
  padding: 0px 4px;
  height: 24px;
  min-width: 24px;
}

.c6:hover {
  background-color: #f1e9dc;
}

.c6:active {
  background-color: #f8f2ea;
}

.c6:focus-visible {
  outline: 2px solid #292924;
  outline-offset: 2px;
}

.c4 {
  background: #ffffff;
  cursor: pointer;
  display: flex;
  justify-content: center;
  border-radius: 24px;
  width: fit-content;
  padding: 0px 4px;
  height: 24px;
  min-width: 24px;
}

.c4:hover {
  background-color: #f1e9dc;
}

.c4:active {
  background-color: #f8f2ea;
}

.c4:focus-visible {
  outline: 2px solid #292924;
  outline-offset: 2px;
}

.c2 {
  cursor: pointer;
}

.c2:focus-visible {
  outline: 2px solid #292924;
  outline-offset: 2px;
}

@media (min-width: 0px) {
  .c0 {
    flex-direction: row;
  }
}

<body>
  <div>
    <div
      class="c0 c1"
    >
      <button
        aria-label="previous-page"
        class="c2"
      >
        <span
          class="c3"
          data-testid="arrow-container"
        >
          <svg
            fill="none"
            height="100%"
            viewBox="0 0 24 24"
            width="100%"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M2.087 12.401c.014.033.042.058.06.089a.99.99 0 00.177.238c.008.007.01.018.018.025l8 7a1 1 0 101.316-1.505L5.661 13H21a1 1 0 100-2H5.661l5.997-5.248a1 1 0 10-1.316-1.504l-8 7c-.008.007-.011.018-.018.025a.99.99 0 00-.177.238c-.018.031-.045.056-.06.089a.97.97 0 000 .801z"
              fill="currentColor"
            />
          </svg>
        </span>
      </button>
      <button
        class="c4"
      >
        <p
          class="c5"
          cursor="inherit"
          title=""
        >
          1
        </p>
      </button>
      <button
        class="c4"
      >
        <p
          class="c5"
          cursor="inherit"
          title=""
        >
          2
        </p>
      </button>
      <button
        class="c4"
      >
        <p
          class="c5"
          cursor="inherit"
          title=""
        >
          3
        </p>
      </button>
      <button
        class="c6"
      >
        <p
          class="c7"
          cursor="inherit"
          title=""
        >
          4
        </p>
      </button>
      <button
        class="c4"
      >
        <p
          class="c5"
          cursor="inherit"
          title=""
        >
          5
        </p>
      </button>
      <button
        class="c4"
      >
        <p
          class="c5"
          cursor="inherit"
          title=""
        >
          ...
        </p>
      </button>
      <button
        class="c4"
      >
        <p
          class="c5"
          cursor="inherit"
          title=""
        >
          12
        </p>
      </button>
      <button
        aria-label="next-page"
        class="c2"
      >
        <span
          class="c8"
          data-testid="arrow-container"
        >
          <svg
            fill="none"
            height="100%"
            viewBox="0 0 24 24"
            width="100%"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M2.087 12.401c.014.033.042.058.06.089a.99.99 0 00.177.238c.008.007.01.018.018.025l8 7a1 1 0 101.316-1.505L5.661 13H21a1 1 0 100-2H5.661l5.997-5.248a1 1 0 10-1.316-1.504l-8 7c-.008.007-.011.018-.018.025a.99.99 0 00-.177.238c-.018.031-.045.056-.06.089a.97.97 0 000 .801z"
              fill="currentColor"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
</body>
`;

exports[`Pagination > renders correctly with default props 1`] = `
.c0 {
  display: flex;
}

.c7 {
  display: flex;
  align-items: center;
  justify-content: center;
  flex-shrink: 0;
  width: 18px;
  height: 18px;
  transform: rotate(180deg);
}

.c7 svg {
  color: #292924!important;
}

.c3 {
  font-size: 14px;
  font-weight: 500;
  line-height: 20px;
  text-align: left;
  color: #292924;
}

.c5 {
  font-size: 16px;
  font-weight: 400;
  line-height: 20px;
  text-align: left;
  color: #292924;
}

.c1 {
  align-items: center;
}

.c1 >button {
  margin-right: 4px;
}

.c1 >button:last-child {
  margin-right: 0;
}

.c2 {
  background: #f1e9dc;
  cursor: pointer;
  display: flex;
  justify-content: center;
  border-radius: 24px;
  width: fit-content;
  padding: 0px 4px;
  height: 24px;
  min-width: 24px;
}

.c2:hover {
  background-color: #f1e9dc;
}

.c2:active {
  background-color: #f8f2ea;
}

.c2:focus-visible {
  outline: 2px solid #292924;
  outline-offset: 2px;
}

.c4 {
  background: #ffffff;
  cursor: pointer;
  display: flex;
  justify-content: center;
  border-radius: 24px;
  width: fit-content;
  padding: 0px 4px;
  height: 24px;
  min-width: 24px;
}

.c4:hover {
  background-color: #f1e9dc;
}

.c4:active {
  background-color: #f8f2ea;
}

.c4:focus-visible {
  outline: 2px solid #292924;
  outline-offset: 2px;
}

.c6 {
  cursor: pointer;
}

.c6:focus-visible {
  outline: 2px solid #292924;
  outline-offset: 2px;
}

@media (min-width: 0px) {
  .c0 {
    flex-direction: row;
  }
}

<body>
  <div>
    <div
      class="c0 c1"
    >
      <button
        class="c2"
      >
        <p
          class="c3"
          cursor="inherit"
          title=""
        >
          1
        </p>
      </button>
      <button
        class="c4"
      >
        <p
          class="c5"
          cursor="inherit"
          title=""
        >
          2
        </p>
      </button>
      <button
        class="c4"
      >
        <p
          class="c5"
          cursor="inherit"
          title=""
        >
          3
        </p>
      </button>
      <button
        class="c4"
      >
        <p
          class="c5"
          cursor="inherit"
          title=""
        >
          4
        </p>
      </button>
      <button
        class="c4"
      >
        <p
          class="c5"
          cursor="inherit"
          title=""
        >
          5
        </p>
      </button>
      <button
        class="c4"
      >
        <p
          class="c5"
          cursor="inherit"
          title=""
        >
          ...
        </p>
      </button>
      <button
        class="c4"
      >
        <p
          class="c5"
          cursor="inherit"
          title=""
        >
          12
        </p>
      </button>
      <button
        aria-label="next-page"
        class="c6"
      >
        <span
          class="c7"
          data-testid="arrow-container"
        >
          <svg
            fill="none"
            height="100%"
            viewBox="0 0 24 24"
            width="100%"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M2.087 12.401c.014.033.042.058.06.089a.99.99 0 00.177.238c.008.007.01.018.018.025l8 7a1 1 0 101.316-1.505L5.661 13H21a1 1 0 100-2H5.661l5.997-5.248a1 1 0 10-1.316-1.504l-8 7c-.008.007-.011.018-.018.025a.99.99 0 00-.177.238c-.018.031-.045.056-.06.089a.97.97 0 000 .801z"
              fill="currentColor"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
</body>
`;
